File: CS/WinForms-Dashboard-State-Changed-event/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;
using System.Windows.Forms;
using System.Xml.Linq;

namespace WinForms_Dashboard_State_Changed_event {
    public partial class Form1: Form {
        DashboardState state = new DashboardState();
        public static readonly string PropertyName = "DashboardState";
        const string path = @"..\..\Dashboards\dashboard1.xml";
        public Form1() {
            InitializeComponent();
            dashboardViewer.LoadDashboard(path);
        }

        private void DashboardViewer_DashboardStateChanged(object sender, DashboardStateChangedEventArgs e) {
            state = dashboardViewer.GetDashboardState();
            var stateValue = state.SaveToXml().ToString(SaveOptions.DisableFormatting);
            dashboardViewer.Dashboard.CustomProperties.SetValue("DashboardState", stateValue);
        }

        private void DashboardViewer_SetInitialDashboardState(object sender, SetInitialDashboardStateEventArgs e) {
            e.InitialState = state;
        }

        void GetDataFromString(string customPropertyValue) {
            if(!string.IsNullOrEmpty(customPropertyValue)) {
                var xmlStateEl = XDocument.Parse(customPropertyValue);
                state.LoadFromXml(xmlStateEl);
            }
        }
        private void DashboardViewer_DashboardLoaded(object sender, DashboardLoadedEventArgs e) {
            GetDataFromString(dashboardViewer.Dashboard.CustomProperties.GetValue(PropertyName));
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e) {
            dashboardViewer.Dashboard.SaveToXml(path);
        }
    }
}

File: VB/WinForms-Dashboard-State-Changed-event/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin
Imports System.Windows.Forms
Imports System.Xml.Linq

Namespace WinForms_Dashboard_State_Changed_event

    Public Partial Class Form1
        Inherits Form

        Private state As DashboardState = New DashboardState()

        Public Shared ReadOnly PropertyName As String = "DashboardState"

        Const path As String = "..\..\Dashboards\dashboard1.xml"

        Public Sub New()
            InitializeComponent()
            dashboardViewer.LoadDashboard(path)
        End Sub

        Private Sub DashboardViewer_DashboardStateChanged(ByVal sender As Object, ByVal e As DashboardStateChangedEventArgs)
            state = dashboardViewer.GetDashboardState()
            Dim stateValue = state.SaveToXml().ToString(SaveOptions.DisableFormatting)
            dashboardViewer.Dashboard.CustomProperties.SetValue("DashboardState", stateValue)
        End Sub

        Private Sub DashboardViewer_SetInitialDashboardState(ByVal sender As Object, ByVal e As SetInitialDashboardStateEventArgs)
            e.InitialState = state
        End Sub

        Private Sub GetDataFromString(ByVal customPropertyValue As String)
            If Not String.IsNullOrEmpty(customPropertyValue) Then
                Dim xmlStateEl = XDocument.Parse(customPropertyValue)
                state.LoadFromXml(xmlStateEl)
            End If
        End Sub

        Private Sub DashboardViewer_DashboardLoaded(ByVal sender As Object, ByVal e As DashboardLoadedEventArgs)
            GetDataFromString(dashboardViewer.Dashboard.CustomProperties.GetValue(PropertyName))
        End Sub

        Private Sub Form1_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs)
            dashboardViewer.Dashboard.SaveToXml(path)
        End Sub
    End Class
End Namespace
